pkg/microservice/aslan/core/common/repository/mongodb/scanning.go

```diff
@@ -0,0 +1,148 @@
+/*
+Copyright 2022 The KodeRover Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package mongodb
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+
+	"github.com/koderover/zadig/pkg/microservice/aslan/config"
+	"github.com/koderover/zadig/pkg/microservice/aslan/core/common/repository/models"
+	mongotool "github.com/koderover/zadig/pkg/tool/mongo"
+)
+
+type ScanningListOption struct {
+	ProjectName string
+}
+
+type ScanningColl struct {
+	*mongo.Collection
+
+	coll string
+}
+
+func NewScanningColl() *ScanningColl {
+	name := models.Scanning{}.TableName()
+	return &ScanningColl{
+		Collection: mongotool.Database(config.MongoDatabase()).Collection(name),
+		coll:       name,
+	}
+}
+
+func (c *ScanningColl) GetCollectionName() string {
+	return c.coll
+}
+
+func (c *ScanningColl) EnsureIndex(ctx context.Context) error {
+	return nil
+}
+
+func (c *ScanningColl) Create(scanning *models.Scanning) error {
+	if scanning == nil {
+		return errors.New("nil testing args")
+	}
+
+	scanning.CreatedAt = time.Now().Unix()
+
+	_, err := c.InsertOne(context.TODO(), scanning)
+	return err
+}
+
+func (c *ScanningColl) Update(idString string, scanning *models.Scanning) error {
+	if scanning == nil {
+		return fmt.Errorf("nil object")
+	}
+	id, err := primitive.ObjectIDFromHex(idString)
+	if err != nil {
+		return fmt.Errorf("invalid id")
+	}
+
+	scanning.UpdatedAt = time.Now().Unix()
+
+	filter := bson.M{"_id": id}
+	update := bson.M{"$set": scanning}
+
+	_, err = c.UpdateOne(context.TODO(), filter, update)
+	return err
+}
+
+func (c *ScanningColl) List(listOption *ScanningListOption, pageNum, pageSize int64) ([]*models.Scanning, int64, error) {
+	query := bson.M{}
+	resp := make([]*models.Scanning, 0)
+	ctx := context.Background()
+
+	opt := options.Find()
+
+	if pageNum != 0 && pageSize != 0 {
+		opt.
+			SetSkip((pageNum - 1) * pageSize).
+			SetLimit(pageSize)
+	}
+
+	if listOption != nil {
+		query["project_name"] = listOption.ProjectName
```

Comment:
In this case, the project name must not be equal to empty?

Reply:
yeah for now you could just pass nil as an argument